compute_tools/src/compute.rs

```diff
@@ -442,8 +442,42 @@ impl ComputeNode {
 
         let pg = self.start_postgres(spec.storage_auth_token.clone())?;
 
+        // Maybe apply the spec
         if spec.spec.mode == ComputeMode::Primary {
-            self.apply_config(&compute_state)?;
+            let spec = &compute_state.pspec.as_ref().expect("spec must be set").spec;
+
+            // Get spec_id or make it up by hashing
+            //
+            // TODO Make spec_id required so there would be no need to hash.
+            let spec_id = spec.operation_uuid.clone().unwrap_or_else(|| {
```

Comment:
Jumping from bellow, I'd say that to make caching properly working we have to always hash. As operations ids could be different for completely identical specs in terms of `apply_conf`